src/Forms/GridField/GridState_Data.php

```diff
@@ -46,7 +65,7 @@ public function getData($name, $default = null)
             $this->data[$name] = $default;
         } else {
             if (is_array($this->data[$name])) {
-                $this->data[$name] = new GridState_Data($this->data[$name]);
+                $this->data[$name] = new self($this->data[$name]);
```

Comment:
Interested to know whether this was "necessary" or just "seemed sensible". Do we know of uses of subclasses?

Reply:
It's more a personal preference. Whenever I have a class that instantiate an instance of itself, I default to using `new self()` so I can copy paste the code other places without having to think about it.

Feel free to switch it back to `new GridState_Data()`

Reply:
I see that subclasses of GridState_Data aren't really part of the design so this is probably relatively innocuous.